util/vizerror/vizerror.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package vizerror provides types and utility funcs for handling visible errors
+// that are safe to display to end users.
+package vizerror
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Error is an error that is safe to display to end users.
+type Error error
+
+// New returns an error that formats as the given text.
+func New(text string) error {
+	return Error(errors.New(text))
+}
+
+// Errorf returns an error with the specified format and values.
+func Errorf(format string, a ...any) error {
+	return Error(fmt.Errorf(format, a...))
+}
```

Comment:
maybe add the `func FromError(err error) (_ Error, ok bool)` too that does the errors.As for caller?

maybe `FromError` is too long. Maybe `Get` or `As` or `From`? :shrug:

Reply:
maybe I just haven't used `errors.As` enough, but why would we use that over simply casting:`vizerror.Error(existingErr)` ?

Reply:
You mean type assert and not cast. But it doesn't work in any case: https://go.dev/play/p/fPJ_y_F_0q1

You don't want the error as the underlying type. Do `type Error struct { error }`

But you can't just type assert because of wrapping. If somebody does `fmt.Errorf("bad policy: %w", err)` you want to be able to get the user visible error out of that %w-wrapped inner `err` too.

Reply:
See https://go.dev/blog/go1.13-errors

Reply:
I added a simple `Wrap(err) Error` func.  But I'm not if that's what you were getting and, since I'm not sure what you mean by doing the errors.As for the caller.  Isn't that going the other direction... unwrapping and getting the `vizerror.Error` back out?